controllers/volsync/vshandler_test.go

```diff
@@ -1837,6 +1838,30 @@ var _ = Describe("VolSync_Handler", func() {
 	})
 })
 
+func retryEnsurePVCfromRD(vsHandler *volsync.VSHandler, rdSpec ramendrv1alpha1.VolSyncReplicationDestinationSpec,
+) error {
+	var err error
+
+	Eventually(func() bool {
+		err = vsHandler.EnsurePVCfromRD(rdSpec, false)
+		if err != nil {
+			if kerrors.IsNotFound(err) {
+				var statusError *kerrors.StatusError
+
+				return errors.As(err, &statusError) &&
+					statusError.ErrStatus.Details != nil &&
+					statusError.ErrStatus.Details.Kind != "PersistentVolumeClaim"
```

Comment:
Why do we care about the details of the not found error?

Trying to extract the underlying error means we should have returned it as is from the caller. It does not make sense to create unhelpful error and then work hard on extracting the helpful details from the error. This should be fixed in vshandler.

Reply:
Here is comment from @ShyamsundarR in the issue itself: https://github.com/RamenDR/ramen/issues/1371#issue-2278950857